src/qbe_ir_generator.cpp

```diff
@@ -151,11 +151,32 @@ void QbeIrGenerator::Visit(const DeclVarNode& decl) {
   id_to_num[decl.id] = id_num;
 }
 
-void QbeIrGenerator::Visit(const DeclArrNode& array_decl) {
-  int id_num = NextLocalNum();
-  WriteInstr_("{} =l alloc{} {}", FuncScopeTemp{id_num},
-              array_decl.type->element_type().size(), array_decl.type->size());
-  id_to_num[array_decl.id] = id_num;
+void QbeIrGenerator::Visit(const DeclArrNode& arr_decl) {
+  int base_addr_num = NextLocalNum();
+  std::size_t element_size = arr_decl.type->element_type().size();
+  WriteInstr_("{} =l alloc{} {}", FuncScopeTemp{base_addr_num}, element_size,
```

Comment:
Since the variable `element_size` is only used once, and `arr_decl.type->element_type().size()` is expressive enough, consider inline the variable:
```suggestion
  std::size_t element_size = arr_decl.type->element_type().size();
  WriteInstr_("{} =l alloc{} {}", FuncScopeTemp{base_addr_num}, arr_decl.type->element_type().size(),
```